### PR TITLE
Add support for the pnpm package manager

### LINKED ISSUE #913: Add support for PNPM
Allow to scan for licenses when using PNPM instead of `yarn` or `npm`

### PR DESCRIPTION
**Pull request by weyert**
Adds support for the PNPM package manager

Depends on a new feature to be merged and published in the PNPM package

fixes #913

### CODE REVIEW

**Comment by xtreme-shane-lattanzio** [comment]
Hey @weyert ! Thanks for the PR. I havent looked but Im guessing the tests are failing because the new package manager was not added to the dockerfile. If it does not need to be added and comes default, we will still need to look into getting rubocop and the tests passing

**Comment by xtreme-shane-lattanzio** [comment]
With respect to the rubocop issues, the version we have been using is 1.12.1. If you are using a different version, you can get messages about additional cops.

For the installation, FYI it looks like you download to a different directory and then dont move it to the local bin folder before trying to chmod `-o /bin/pnpm; chmod +x /usr/local/bin/pnpm`

**Comment by xtreme-shane-lattanzio** [comment]
Looks like the package manager is now installed! Errors now seem to be about undefined variables when doing a pnpm version check

**Comment by weyert** [comment]
> Looks like the package manager is now installed! Errors now seem to be about undefined variables when doing a pnpm version check

Yeah slowly getting there. Having trouble to run the tests locally so misusing this PR to get it green. Think I forgot to add the pnpm lockfile to the `all_pms` spec fixture and rubocop wanted me to add a `?` to a function but forgot to apply the changes were the function was used. Let's how happy the CI is now :D

File: features/support/testing_dsl.rb [hunk 1]
```diff
@@ -223,6 +223,13 @@ def add_dep
       end
     end
 
+    class PNPMProject < Project
+      def add_dep
+        add_to_file('pnpm-lock.yaml', '')
+        add_to_file('package.json', '{"dependencies" : {"http-server": "0.11.1"}}')
```
Comment (xtreme-shane-lattanzio) [inline features/support/testing_dsl.rb:229]: This test is failing. Should you be adding this to package.json or to the lock file?

**Comment by weyert** [comment]
Still interested in finishing this PR. Waiting for PNPMs next release; the licenses command got recently merged

**Comment by xtreme-shane-lattanzio** [comment]
Hey! Sounds good. Im looking to create a release this week and it would be great if we can get this in. It looks like the failures are now about minimum pnpm versions not being met. Hopefully it is an easy fix!

**Comment by weyert** [comment]
> Hey! Sounds good. Im looking to create a release this week and it would be great if we can get this in. It looks like the failures are now about minimum pnpm versions not being met. Hopefully it is an easy fix!

Yeah, I will have a look tonight. I just wished I was able to run the tests on my Mac. Somehow running the tests against this PRs takes a long time. Do you need to manual approve them?

**Comment by xtreme-shane-lattanzio** [comment]
Manually approve running tests? No. Im not sure why you would have an issue but if you arent using the LF docker image, I'd start there. If you are then im not sure but it must. be some kind of environment issue

**Comment by weyert** [comment]
Yeah my understanding of running the tests in the docker image is I just run `./dlf rake spec` in this repo's root and then th e tests should run but then I am getting:

```bash
Status: Downloaded newer image for licensefinder/license_finder:latest
WARNING: The requested image's platform (linux/amd64) does not match the detected host platform (linux/arm64/v8) and no specific platform was requested
/usr/share/rvm/rubies/ruby-3.1.1/bin/ruby -I/usr/share/rvm/gems/ruby-3.1.1/gems/rspec-support-3.11.0/lib:/usr/share/rvm/gems/ruby-3.1.1/gems/rspec-core-3.11.0/lib /usr/share/rvm/gems/ruby-3.1.1/gems/rspec-core-3.11.0/exe/rspec --pattern ./spec/\*\*/\*_spec.rb --color
Could not find addressable-2.8.1, fakefs-1.8.0, rspec-3.12.0, webmock-3.18.1, nokogiri-1.13.9, rack-3.0.0, rack-test-2.0.2, tomlrb-2.0.3, public_suffix-4.0.7, rspec-core-3.12.0, rspec-expectations-3.12.0, rspec-mocks-3.12.0, parallel-1.22.1, parser-3.1.2.1, rubocop-ast-1.23.0, unicode-display_width-2.3.0, mini_portile2-2.8.0, rspec-support-3.12.0 in any of the sources
Run `bundle install` to install missing gems.

While loading ./spec/lib/license_finder/cli/approvals_spec.rb an `exit` / `raise SystemExit` occurred, RSpec will now quit.
Failure/Error: require 'bundler/setup'
```

**Comment by weyert** [comment]
@xtreme-shane-lattanzio I think its ready for review

**Comment by weyert** [comment]
Test is failing but for an unrelated unit test?
>  Failure/Error: LicenseFinder::TestingDSL::BowerProject.create

**Comment by xtreme-shane-lattanzio** [comment]
Hey @weyert ! Thanks for sticking with this! It looks good so it will be in the next release!

**Comment by weyert** [comment]
Thank you @xtreme-shane-lattanzio  for accepting and merging my pull request :)

**Pull request merged**